server/channels/api4/brand.go

```diff
@@ -23,16 +23,27 @@ func getBrandImage(c *Context, w http.ResponseWriter, r *http.Request) {
 	img, err := c.App.GetBrandImage(c.AppContext)
 	if err != nil {
 		w.WriteHeader(http.StatusNotFound)
-		w.Write(nil)
+		if _, writeErr := w.Write(nil); writeErr != nil {
+			c.Err = model.NewAppError("getBrandImage", "api.response_writer_error", nil, "", http.StatusInternalServerError)
+			return
+		}
```

Comment:
If the write fails, there is no point in sending another error to the client. It will likely also fail. Let's just log an error and be done. You can follow the pattern from 
https://github.com/mattermost/mattermost/blob/f1903754034b395b1fb0aef337ed689b17e3d61a/server/channels/api4/system.go#L816-L818